app/uk/gov/hmrc/cgtpropertydisposalsfrontend/views/returns/task_list.scala.html

```diff
@@ -34,6 +39,39 @@
 @propertyAddressState = @{ draftReturn.propertyAddress.fold[TaskListStatus](TaskListStatus.ToDo)(_ => TaskListStatus.Complete) }
 @disposalDetailsState = @{ draftReturn.disposalDetailsAnswers.fold[TaskListStatus](TaskListStatus.ToDo)(_.fold(_ => TaskListStatus.InProgress,_ => TaskListStatus.Complete)) }
 @acquistionDetailsState = @{ draftReturn.acquisitionDetailsAnswers.fold[TaskListStatus](TaskListStatus.ToDo)(_.fold(_ => TaskListStatus.InProgress,_ => TaskListStatus.Complete)) }
+
+@initialGainOrLossState = @{
+    def canUseService = {
+        val residentialBeforeDate = LocalDateUtils.dateFromString("1/04/2015")
+        val nonResidentialBeforeDate = LocalDateUtils.dateFromString("1/04/2019")
+        (triageState) match {
+            case (TaskListStatus.Complete) => {
+                val CompleteSingleDisposalTriageAnswers(_, _, countryOfResidence, assetType, _, _) = draftReturn.triageAnswers
```

Comment:
this isn't a 100% safe way of casting the triage answers as `CompleteSingleDisposalTriageAnswers` - you're inferring that it should be complete from the outer pattern match. It that was incorrect then you'd end up with a match error. What would be safer is to match on `draftReturn.triageAnswers` itself to make sure that what you have is a complete version of the answers.

Reply:
Done.